codecov_cli/services/report/__init__.py

```diff
@@ -27,21 +27,33 @@ def create_report_logic(
     service: str,
     token: uuid.UUID,
     enterprise_url: str,
+    pull_request_number: int,
     fail_on_error: bool = False,
 ):
     encoded_slug = encode_slug(slug)
     sending_result = send_create_report_request(
-        commit_sha, code, service, token, encoded_slug, enterprise_url
+        commit_sha,
+        code,
+        service,
+        token,
+        encoded_slug,
+        enterprise_url,
+        pull_request_number,
     )
     log_warnings_and_errors_if_any(sending_result, "Report creating", fail_on_error)
     return sending_result
 
 
 def send_create_report_request(
-    commit_sha, code, service, token, encoded_slug, enterprise_url
+    commit_sha, code, service, token, encoded_slug, enterprise_url, pull_request_number
 ):
     data = {"code": code}
-    headers = get_token_header_or_fail(token)
+    decoded_slug = decode_slug(encoded_slug)
+    headers = (
+        {}
+        if not token and is_fork_pr(pull_request_number, decoded_slug, service)
+        else get_token_header_or_fail(token)
+    )
```

Comment:
super minor nit: when you merge, could you mention this extra little change in the commit message? just in case someone is reading through commit history searching for a bug or trying to learn how tokenless was implemented. something like `adding PR option and tokenless logic to create-report command`